### PR TITLE
feat(gatsby-source-wordpress): Add gatsby plugin image

### PR DESCRIPTION
**Pull request by DanielSLew**
### Documentation

Replacing the previous `gatsby-image` with `gatsby-plugin-image`.

Because of the way that the WP/source plugin handles images (converting to raw html string, rather than React) we needed to explicitly hydrate each image created.

### CODE REVIEW

**Comment by wardpeet** [comment]
We have to wait until tomorrow to merge, so we're on v4. Cause introducing gatsby-plugin-image now would mean a breaking change as it might break some layouts subtle.

File: packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js:614]:
```suggestion
                ...(pluginOptions?.html?.gatsbyImageOptions || {}),
```

I think if this isn't an object it'll throw

File: packages/gatsby-source-wordpress/src/gatsby-browser.ts
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/src/gatsby-browser.ts:10]: Does this work when doing a client-side navigation to another route? Wont it only hydrate the very first page visited?

**Review by TylerBarnes — CHANGES_REQUESTED** [review]

File: packages/gatsby-source-wordpress/gatsby-browser.js
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/gatsby-browser.js:1]: I think this might be a breaking change for users who relied on these styles existing and added their own styles on top. @wardpeet will this change just show up in the v6 version or does this come out in a minor?

Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/gatsby-browser.js:1]: Oh, just saw your other comment. 👌

Comment (wardpeet) [reply packages/gatsby-source-wordpress/gatsby-browser.js:1]: We won't do any 3.x releases except hotfixes but they are in a different branch

File: packages/gatsby-source-wordpress/src/gatsby-browser.ts
Comment (wardpeet) [reply packages/gatsby-source-wordpress/src/gatsby-browser.ts:10]: It doens't hydrate on first pass. Only on client routes - gatsby-plugin-image is smart and takes care of the first pass without react

File: packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js
Comment (wardpeet) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js:614]: Perhaps we should add default {} to pluginOptions above so it's always set

Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js:614]: @wardpeet it can be set with a default but a user could make pluginOptions.html false or null manually I believe

File: packages/gatsby-source-wordpress/src/gatsby-browser.ts
Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/gatsby-browser.ts:10]: Ah, got it 👍 I read the code as if it worked opposite. Thanks for the explanation!

File: packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js
Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js:614]: Actually the user plugin options processors can add a default option https://github.com/gatsbyjs/gatsby/blob/87b70010409df5ff5f199e2a2ac37b15e77cba4f/packages/gatsby-source-wordpress/src/steps/process-and-validate-plugin-options.ts#L74

Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js:614]: You can add it here more easily https://github.com/gatsbyjs/gatsby/blob/87b70010409df5ff5f199e2a2ac37b15e77cba4f/packages/gatsby-source-wordpress/src/models/gatsby-api.ts#L174 but the user can change it I think

Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/process-node.js:614]: Just added a default. I think it's pretty edge casey for a user to add html: null and if they do the build will probably fail immediately and they'll undo that change. So it should be fine :)

**Review by TylerBarnes — APPROVED** [review]

**Comment by TylerBarnes** [comment]
Woo! Related tests are passing 🎉

**Comment by TylerBarnes** [comment]
Thanks for this @DanielSLew and @wardpeet ! 💪

**Pull request merged**
